vars/buildDockerAndPublishImage.groovy

```diff
@@ -14,6 +14,7 @@ def call(String imageShortName, Map userConfig=[:]) {
     gitCredentials: 'github-app-infra', // Credential ID for tagging and creating release
     imageDir: '.', // Relative path to the context directory for the Docker build
     registryNamespace: '', // Empty by default (means "autodiscover based on the current controller")
+    unstash: '', // Allow to unstash files if not empty
```

Comment:
Warning here: I'll approve the PR assuming you've tested it on real life and that it unblocks https://github.com/jenkins-infra/plugin-health-scoring/pull/36.

BUT: be aware that the `unstash` pipeline instruction can be specified with a map of arguments (instead of the short call with a string as sole argument).

Maybe worth setting the default value to empty and use Groovy magic (`if (finalConfig.unstash) { ...}`) so it would be transparent?

Better adding unit test for both cases

Reply:
yes, my apologies I mis-wrote. `null` looks good (assuming you can write it with Groovy). don't forget to test in both "unit test execution engine" environement, and also inside a fully-fledged Jenkins instance.

Reply:
> be aware that the unstash pipeline instruction can be specified with a map of arguments (instead of the short call with a string as sole argument).

Do you have a reference to this map of args? I've only found this one saying the `unstash` command accepts a string: https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#unstash-restore-files-previously-stashed

Reply:
Ooooh you are correct, I mixed up with `unstash`.

Under the hood, the `unstash` keywrod can style accept a map with the key `name` of type String (as per the documentation). But it's unlikely that another would be added so you should go with the initial proposal with an empty String 👍